pandas/core/series.py

```diff
@@ -235,6 +234,18 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                 if not is_list_like(data):
                     data = [data]
                 index = com._default_index(len(data))
+            elif is_list_like(data):
+
+                # a scalar numpy array is list-like but doesn't
+                # have a proper length
+                try:
+                    if len(data) > 1 and len(index) != len(data):
```

Comment:
hmm, did this change? 0-len should be ok, can you add a test

Reply:
0-len gets caught deeper, in the SingleBlockManager, but len 1 gets caught here. It should be let through to be broadcast in `_sanitize_array`. I'll add a test.

Reply:
hmm, would be ok with catching both cases here, or are they different?

Reply:
Yeah. I think catching the 0-len case here would be good for consistency. We don't want to catch the len 1 case because it will get broadcast, so it will look something like:
```
                # a scalar numpy array is list-like but doesn't
                # have a proper length
                try:
                    if len(data) != 1 and len(index) != len(data):
```
Unless the intention is not to broadcast a list-like of length 1.  One could argue that it would be better to raise an error instead of broadcasting.  If one wanted to broadcast a scalar, then just pass a scalar.

Reply:
you would have to show the test which fails for this, len(data) == 0 is valid

Reply:
The test `test_apply_subset` in tests/io/formats/test_style.py raises an error. The traceback indicates the input to the Series constructor is:
```
data = ['color: baz'], index = RangeIndex(start=0, stop=2, step=1), dtype = None
```
This should be valid.  Checking that `len(data) != 1` lets this case pass.